stdlib/asm/math/u64.masm

```diff
@@ -156,8 +161,26 @@ end
 # Stack transition looks as follows:
 # [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a * b) % 2^64
 export.checked_mul
-    exec.u32assert4
-    exec.overflowing_mul
+    dup.3
+    dup.2
+    u32assert.2
+    u32overflowing_mul
+    dup.4
+    movup.4
+    u32overflowing_madd
+    swap
+    movup.5
+    dup.4
+    u32overflowing_madd
+    movup.5
+    movup.5
+    u32assert.2
+    u32overflowing_madd
+    movup.3
+    movup.2
+    u32overflowing_add
+    movup.2
+    add
     u32checked_or
     eq.0
```

Comment:
`u32checked_or` is relatively expensive. I think we should be able to replace it with a regular `or` (but double check me on this).

Also, my understanding is that the last few instructions are just checking if the top 3 values on the stack are zeros. And I think we know that all of them are smaller than $2^32$ - so, maybe we can do something a bit more efficient?

For example, could we replace:
```
movup.2
add
u32checked_or
eq.0
assert
```
with
```
add
add
eq.0
assert
```

Reply:
Thanks. I think it makes sense. Changed the sequence of instructions with your snippet.